tests/src/Kernel/TranslationsRevisionTest.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_translation\Kernel;
+
+use Drupal\KernelTests\KernelTestBase;
+use Drupal\oe_translation\TranslationModerationHandler;
+use Drupal\Tests\content_moderation\Traits\ContentModerationTestTrait;
+use Drupal\Tests\user\Traits\UserCreationTrait;
+
+/**
+ * Tests that whenever a new translation is saved, no revision is created.
+ *
+ * This is needed whenever the site uses content moderation which forces a
+ * new revision every time the node is saved.
+ */
+class TranslationsRevisionTest extends KernelTestBase {
+
+  use ContentModerationTestTrait;
+  use UserCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'path',
+    'field',
+    'options',
+    'text',
+    'node',
+    'user',
+    'system',
+    'filter',
+    'workflows',
+    'language',
+    'oe_multilingual',
+    'content_moderation',
+    'content_translation',
+    'views',
+    'path_alias',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected $defaultTheme = 'classy';
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp(): void {
+    parent::setUp();
+
+    $this->installConfig([
+      'system',
+      'node',
+      'field',
+      'user',
+      'workflows',
+      'language',
+      'oe_multilingual',
+      'content_moderation',
+    ]);
+
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('user');
+    $this->installEntitySchema('content_moderation_state');
+    $this->installEntitySchema('path_alias');
+
+    $this->installSchema('node', 'node_access');
+    $this->installSchema('system', 'sequences');
+
+    $node_type = $this->container->get('entity_type.manager')->getStorage('node_type')->create([
+      'name' => 'Test node type',
+      'type' => 'test_node_type',
+    ]);
+
+    $node_type->save();
+
+    $workflow = $this->createEditorialWorkflow();
+    $workflow->getTypePlugin()
+      ->addEntityTypeAndBundle('node', 'test_node_type');
+
+    $workflow->save();
+  }
+
+  /**
+   * Tests that no revisions are created when saving a translation.
+   *
+   * This happens when a node type uses a content moderation workflow and we
+   * prevent new revisions from being created when a new translation is saved.
+   */
+  public function testNoRevisionIncrement(): void {
+    /** @var \Drupal\node\NodeStorageInterface $storage */
+    $storage = $this->container->get('entity_type.manager')->getStorage('node');
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $storage->create([
+      'type' => 'test_node_type',
+      'title' => 'Revisions',
+    ]);
+
+    $node->save();
```

Comment:
Can we replace with `createBasicTestNode` from the new Trait for the first part?